File: WEEK_3_C/Q3.c
#include <stdio.h>
#include <stdlib.h>

struct Node{ int data; struct Node* next; };

void print(struct Node* head){
    while(head){ printf("%d ",head->data); head=head->next; }
    printf("\n");
}

int main(){
    struct Node *head=NULL,*t,*prev;
    for(int i=1;i<=3;i++){ t=(struct Node*)malloc(sizeof(struct Node)); t->data=i; t->next=head; head=t; }
    int key=2; prev=NULL; t=head;
    while(t && t->data!=key){ prev=t; t=t->next; }
    if(t){ if(prev) prev->next=t->next; else head=t->next; free(t); }
    print(head);
    return 0;
}

File: WEEK_3_Python/Q3.py
class Node:
    def __init__(self, data):
        self.data, self.next = data, None

def print_list(head):
    cur = head
    while cur:
        print(cur.data, end=" ")
        cur = cur.next
    print()

def delete_node(head, key):
    if head and head.data == key:
        return head.next
    cur = head
    while cur and cur.next:
        if cur.next.data == key:
            cur.next = cur.next.next
            break
        cur = cur.next
    return head

# Example
head = Node(1); head.next = Node(2); head.next.next = Node(3)
head = delete_node(head, 2)
print_list(head)   
